impress-org/givewp
# Feature: add header image overlay setting

Resolves [GIVE-396]

## Description
This pull request introduces an overlay setting that can be applied to the header image of the Visual Form Builder. The setting is accessible from the Styles tab in the Design Settings sidebar and is applicable to all form templates and selected image styles. With this setting, users can select a color from the WordPress palette and adjust the opacity to their preference.

Since the HeaderImage is an image tag, we can not reliably create an overlay with a pseudo element. To achieve the overlay an additional div was included in the HeaderImage template. The settings for both the color & opacity are then passed to the overlay as CSS variables.

## Affects
Visual Form Builder
Header Image

## Visuals

https://github.com/impress-org/givewp/assets/75056371/7efd1f4c-8983-4e8d-a70c-0d2cbed827c3

## Testing Instructions

- Create a form with the VFB & upload a headerImage.
- Select the "Style" tab and click the overlay setting to set a color.
- After a color is chosen you should see the opacity setting. 
- Using the range control slider adjust that the opacity changes accordingly.
- Using the number input update the opacity again.
- Test with all header image styles "background, center, above" & with all available templates.

## Pre-review Checklist



[GIVE-396]: https://stellarwp.atlassian.net/browse/GIVE-396?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
